pkg/nodelabels/builder.go

```diff
@@ -67,13 +67,17 @@ func BuildNodeLabels(cluster *kops.Cluster, instanceGroup *kops.InstanceGroup) m
 		if isAPIServer || featureflag.APIServerNodes.Enabled() {
 			nodeLabels[RoleLabelAPIServer16] = ""
 		}
-		nodeLabels[RoleLabelName15] = RoleAPIServerLabelValue15
+		if cluster.IsKubernetesLT("1.24") {
+			nodeLabels[RoleLabelName15] = RoleAPIServerLabelValue15
+		}
 	} else {
 		if nodeLabels == nil {
 			nodeLabels = make(map[string]string)
 		}
 		nodeLabels[RoleLabelNode16] = ""
-		nodeLabels[RoleLabelName15] = RoleNodeLabelValue15
+		if cluster.IsKubernetesLT("1.24") {
+			nodeLabels[RoleLabelName15] = RoleNodeLabelValue15
+		}
```

Comment:
Is kubeadm also removing the `kubernetes.io/role: node` node label?

Reply:
I don't think kubeadm used that label. https://github.com/kubernetes/kubernetes/pull/107533 does not mention it at all.

Reply:
yea the only reference I can find upstream is this function which handles the `node-role.kubernetes.io/` case as well:

https://github.com/kubernetes/kubernetes/blob/4cdeab4696e86c9738d99e2650d2cdfdfb0f8d32/pkg/printers/internalversion/printers.go#L1653-L1671

Reply:
Note that we have deprecated those labels explicitly so removing these is expected.